src/libs/ReportUtils.ts

```diff
@@ -1536,20 +1541,18 @@ function requiresAttentionFromCurrentUser(option: OnyxEntry<Report> | OptionData
         return false;
     }
 
-    if (isArchivedRoom(option)) {
-        return false;
-    }
-
-    if (isArchivedRoom(getReport(option.parentReportID))) {
+    if (isArchivedRoom(option) || isArchivedRoom(getReport(option.parentReportID))) {
         return false;
     }
 
-    if (Boolean('isUnreadWithMention' in option && option.isUnreadWithMention) || isUnreadWithMention(option)) {
-        return true;
-    }
+    if (isOption(option)) {
+        if (isUnreadWithMention(option)) {
+            return true;
+        }
 
-    if (isWaitingForAssigneeToCompleteTask(option, parentReportAction)) {
-        return true;
+        if (isWaitingForAssigneeToCompleteTask(option, parentReportAction)) {
```

Comment:
Looks like `isWaitingForAssigneeToCompleteTask` expects the first param to be report not an option. 

https://github.com/Expensify/App/pull/32140/files#diff-577fcc5f3a5e4930916ff310e61c38edfa093792fe3ab55b9a30d2aecd8811dbR1511

Reply:
Restricting the use of `isWaitingForAssigneeToCompleteTask` will change order of the report as `requiresAttentionFromCurrentUser` is being used in SidebarUtils to get ordered reportIDs (`getOrderedReportIDs`).  This is the comparison between staging and PR.

**Staging**

https://github.com/Expensify/App/assets/25876548/09e66b2e-d2e2-46b4-b22f-7867b09e4b16

**PR**

https://github.com/Expensify/App/assets/25876548/3d7fa422-1f26-42ce-81fa-0f815274d473

Reply:
Yes, this is a correct behaviour, this one related to this [issue](https://github.com/Expensify/App/issues/30798) which this PR should fix!

Reply:
We can notice change in other chats with green dot indicator as well.

Reply:
Thanks for the update @waterim.